lib/action_view/helpers/dynamic_form.rb

```diff
@@ -11,14 +11,18 @@ module Helpers
     # is a great way of making the record quickly available for editing, but likely to prove lackluster for a complicated real-world form.
     # In that case, it's better to use the +input+ method and the specialized +form+ methods in link:classes/ActionView/Helpers/FormHelper.html
     module DynamicForm
+      DEFAULT_FIELD_OPTION_MAXLENGTH = 30
       # Returns a default input tag for the type of object returned by the method. For example, if <tt>@post</tt>
       # has an attribute +title+ mapped to a +VARCHAR+ column that holds "Hello World":
       #
       #   input("post", "title")
-      #   # => <input id="post_title" name="post[title]" size="30" type="text" value="Hello World" />
-      def input(record_name, method, options = {})
-        raise_broken_code_error
-        InstanceTag.new(record_name, method, self).to_tag(options)
+      #   # => <input id="post_title" name="post[title]" size="30" maxlength="30" type="text" value="Hello World" />
+      #
+      #   input("post", "title", "maxlength" => 10)
+      #   # => <input id="post_title" name="post[title]" size="10" maxlength="10" type="text" value="Hello World" />
+      def input(object_name, method, options = {})
+        options["maxlength"] = DEFAULT_FIELD_OPTION_MAXLENGTH unless options.key?("maxlength")
+        Tags::TextField.new(object_name, method, self, options).render
```

Comment:
the `render` methods sets the `size` attribute based on the `maxlength` option https://github.com/rails/rails/blob/main/actionview/lib/action_view/helpers/tags/text_field.rb#L11-L18